### DISCUSSION TITLE
Grammarly on Zim Wiki

### DISCUSSION THREAD

**Discussion by GottlobFrege** [zim-desktop-wiki/zim-desktop-wiki#2263] [General]
Hi, 
Is there any way to get the program "Grammarly" to work on Zim? As a non-native speaker, I try to take some of my notes in English, but unfortunately, Grammarly (which I pay for and which otherwise does an excellent job) does not work on Zim. It would be great if someone could help me with this.

**Comment by sojusnik** [comment]
An alternative would be [LanguageTool](https://languagetool.org/), which is included for free in LibreOffice. Maybe it could be included in Zim as well?

**Comment by jaap-karssenberg** [comment]
We could contemplate plugins for these tools, but only if you can demonstrate a workable interface to call these tools from a python interface. Without such interface you better first file requests on the other end to make interfaces available.

**Reply by nbCloud91** [reply]
There exists:
1. [language_tool_python](https://github.com/jxmorris12/language_tool_python):
> This is a Python wrapper for [LanguageTool](https://languagetool.org/). LanguageTool is open-source grammar tool, also known as the spellchecker for OpenOffice. This library allows you to make to detect grammar errors and spelling mistakes through a Python script or through a command-line interface.
2. [pyLanguagetool](https://github.com/Findus23/pyLanguagetool)
> A python library and CLI for the LanguageTool JSON API.

**Reply by GottlobFrege** [reply]
Thank you very much for the answer. I am really very sorry. But I don't know these technical things at all (which is why I couldn't understand jaap-karssenberg's answer above). 
I don't want to be a pain, but could you give me a quick tutorial on what exactly I have to do to get the language control tool working on ZimWiki.

**Reply by nbCloud91** [reply]
There is no direct way to use LanguageTool right away as it is from Zim, some kind of plugin has to be written to get the functionality. Both the tools I linked above provide a python interface to LT server (local or remote). A very crude and simple way to check grammar may be made using the "Custom-Tools" feature and hooking it to pyLanguagetool CLI, but to manipulate the text display within Zim, someone would have to write a plugin.

**Reply by sojusnik** [reply]
Here's an easy way to install a local (fully offline) [standalone server of LanguageTool](https://github.com/sonnyp/Eloquent), which then could be used with a Zim plugin.
